File: src/test/java/mc/sbm/simphonycloud/domain/FactureDetailTestSamples.java
package mc.sbm.simphonycloud.domain;

import java.util.Random;
import java.util.UUID;
import java.util.concurrent.atomic.AtomicInteger;

public class FactureDetailTestSamples {

    private static final Random random = new Random();
    private static final AtomicInteger intCount = new AtomicInteger(random.nextInt() + (2 * Short.MAX_VALUE));

    public static FactureDetail getFactureDetailSample1() {
        return new FactureDetail()
            .id(1)
            .factureEnTeteRef(1)
            .numLigne(1)
            .detailType("detailType1")
            .numSiege("numSiege1")
            .quantiteAffiche(1)
            .totalLigne(1)
            .codeRaisonRef(1)
            .multiplicateur(1)
            .referenceInfo("referenceInfo1")
            .referenceInfo2("referenceInfo21")
            .partieDeJourneeRef(1)
            .periodeDeServiceRef(1)
            .numChrono(1)
            .parentFactureDetailRef(1)
            .taxePourcentage(1)
            .taxeMontant(1)
            .modePaiementTotal(1)
            .prix(1)
            .transactionEmployeRef(1)
            .transfertEmployeRef(1)
            .managerEmployeRef(1)
            .repasEmployeRef(1)
            .remiseRef(1)
            .remiseElementMenuRef(1)
            .commissionServiceRef(1)
            .modePaiementRef(1)
            .elementMenuRef(1);
    }

    public static FactureDetail getFactureDetailSample2() {
        return new FactureDetail()
            .id(2)
            .factureEnTeteRef(2)
            .numLigne(2)
            .detailType("detailType2")
            .numSiege("numSiege2")
            .quantiteAffiche(2)
            .totalLigne(2)
            .codeRaisonRef(2)
            .multiplicateur(2)
            .referenceInfo("referenceInfo2")
            .referenceInfo2("referenceInfo22")
            .partieDeJourneeRef(2)
            .periodeDeServiceRef(2)
            .numChrono(2)
            .parentFactureDetailRef(2)
            .taxePourcentage(2)
            .taxeMontant(2)
            .modePaiementTotal(2)
            .prix(2)
            .transactionEmployeRef(2)
            .transfertEmployeRef(2)
            .managerEmployeRef(2)
            .repasEmployeRef(2)
            .remiseRef(2)
            .remiseElementMenuRef(2)
            .commissionServiceRef(2)
            .modePaiementRef(2)
            .elementMenuRef(2);
    }

    public static FactureDetail getFactureDetailRandomSampleGenerator() {
        return new FactureDetail()
            .id(intCount.incrementAndGet())
            .factureEnTeteRef(intCount.incrementAndGet())
            .numLigne(intCount.incrementAndGet())
            .detailType(UUID.randomUUID().toString())
            .numSiege(UUID.randomUUID().toString())
            .quantiteAffiche(intCount.incrementAndGet())
            .totalLigne(intCount.incrementAndGet())
            .codeRaisonRef(intCount.incrementAndGet())
            .multiplicateur(intCount.incrementAndGet())
            .referenceInfo(UUID.randomUUID().toString())
            .referenceInfo2(UUID.randomUUID().toString())
            .partieDeJourneeRef(intCount.incrementAndGet())
            .periodeDeServiceRef(intCount.incrementAndGet())
            .numChrono(intCount.incrementAndGet())
            .parentFactureDetailRef(intCount.incrementAndGet())
            .taxePourcentage(intCount.incrementAndGet())
            .taxeMontant(intCount.incrementAndGet())
            .modePaiementTotal(intCount.incrementAndGet())
            .prix(intCount.incrementAndGet())
            .transactionEmployeRef(intCount.incrementAndGet())
            .transfertEmployeRef(intCount.incrementAndGet())
            .managerEmployeRef(intCount.incrementAndGet())
            .repasEmployeRef(intCount.incrementAndGet())
            .remiseRef(intCount.incrementAndGet())
            .remiseElementMenuRef(intCount.incrementAndGet())
            .commissionServiceRef(intCount.incrementAndGet())
            .modePaiementRef(intCount.incrementAndGet())
            .elementMenuRef(intCount.incrementAndGet());
    }
}

File: src/main/webapp/app/entities/facture-detail/facture-detail.test-samples.ts
import dayjs from 'dayjs/esm';

import { IFactureDetail, NewFactureDetail } from './facture-detail.model';

export const sampleWithRequiredData: IFactureDetail = {
  id: 4158,
};

export const sampleWithPartialData: IFactureDetail = {
  id: 16182,
  utcDateTime: dayjs('2025-05-26T13:37'),
  numSiege: 'with colorful brand',
  estNul: true,
  multiplicateur: 28414,
  referenceInfo: 'highlight colonialism',
  partieDeJourneeRef: 18697,
  numChrono: 26979,
  parentFactureDetailRef: 19599,
  modePaiementTotal: 6800,
  prix: 16874,
  remiseRef: 30432,
  remiseElementMenuRef: 27199,
  modePaiementRef: 10586,
  elementMenuRef: 23715,
};

export const sampleWithFullData: IFactureDetail = {
  id: 3914,
  factureEnTeteRef: 28179,
  numLigne: 16869,
  detailType: 'but',
  utcDateTime: dayjs('2025-05-26T12:26'),
  lclDateTime: dayjs('2025-05-26T05:44'),
  numSiege: 'ick modulo',
  niveauPrix: 5155.77,
  totalAffiche: 2190.62,
  quantiteAffiche: 29320,
  estErreur: true,
  estNul: false,
  estRetourne: false,
  estInvisible: true,
  totalLigne: 29585,
  codeRaisonRef: 13122,
  multiplicateur: 25233,
  referenceInfo: 'observe',
  referenceInfo2: 'judgementally helpfully',
  partieDeJourneeRef: 27372,
  periodeDeServiceRef: 29951,
  numChrono: 22667,
  parentFactureDetailRef: 1676,
  taxePourcentage: 27520,
  taxeMontant: 24509,
  modePaiementTotal: 25585,
  prix: 5673,
  transactionEmployeRef: 5368,
  transfertEmployeRef: 28009,
  managerEmployeRef: 30014,
  repasEmployeRef: 6410,
  remiseRef: 29223,
  remiseElementMenuRef: 23932,
  commissionServiceRef: 11580,
  modePaiementRef: 19907,
  elementMenuRef: 17728,
};

export const sampleWithNewData: NewFactureDetail = {
  id: null,
};

Object.freeze(sampleWithNewData);
Object.freeze(sampleWithRequiredData);
Object.freeze(sampleWithPartialData);
Object.freeze(sampleWithFullData);
